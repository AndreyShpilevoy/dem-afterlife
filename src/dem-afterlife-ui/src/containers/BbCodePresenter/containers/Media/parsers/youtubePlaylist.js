const youtubePlaylist = sourceLink => {
  const parsedLink = sourceLink.match(
    /youtube(?:-nocookie)?\.com\/(?:playlist\?list=|embed\/videoseries\?list=)([-_\w\d]+)/i
  );
  if (parsedLink) {
    return {
      type: "iframe",
      success: true,
      url: `https://www.youtube.com/embed/videoseries?list=${parsedLink[1]}`,
      isShortHeight: false
    };
  }
  return { success: false };
};

export default youtubePlaylist;
