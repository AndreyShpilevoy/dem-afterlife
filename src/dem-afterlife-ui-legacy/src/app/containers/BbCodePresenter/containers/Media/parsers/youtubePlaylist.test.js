/* eslint-disable react/jsx-filename-extension, no-undef, fp/no-unused-expression, fp/no-nil, max-statements */

import youtubePlaylist from './youtubePlaylist';

describe('youtubePlaylist', () => {
    it('should return valid result if pass link', () => {
        expect(youtubePlaylist('https://www.youtube.com/playlist?list=PL7y5d3VOW3rg6nn0bXii79-92jmLYechP')).toMatchSnapshot();
    });

    it('should return unsuccessful result if pass wrong string', () => {
        expect(youtubePlaylist('fake')).toMatchSnapshot();
    });
});
