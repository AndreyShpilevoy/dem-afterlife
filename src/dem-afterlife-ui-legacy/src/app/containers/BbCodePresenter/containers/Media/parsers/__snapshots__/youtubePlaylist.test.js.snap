// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`youtubePlaylist should return unsuccessful result if pass wrong string 1`] = `
Object {
  "success": false,
}
`;

exports[`youtubePlaylist should return valid result if pass link 1`] = `
Object {
  "isShortHeight": false,
  "success": true,
  "type": "iframe",
  "url": "https://www.youtube.com/embed/videoseries?list=PL7y5d3VOW3rg6nn0bXii79-92jmLYechP",
}
`;
